src/utils/sources-tree/tests/addToTree.spec.js

```diff
@@ -239,5 +239,45 @@ describe("sources-tree", () => {
       sources.forEach(source => addToTree(tree, source, domain));
       expect(formatTree(tree)).toMatchSnapshot();
     });
+
+    it("uses projectRoot to filter the list", () => {
+      const testData = [
+        {
+          url: "http://example.com/components/TodoTextInput.js"
+        },
+        {
+          url: "http://example.com/components/Header.js"
+        },
+        {
+          url: "http://example.com/reducers/index.js"
+        },
+        {
+          url: "http://example.com/components/TodoItem.js"
+        },
+        {
+          url: "resource://gre/modules/ExtensionContent.jsm"
+        },
+        {
+          url:
+            "https://voz37vlg5.codesandbox.io/static/js/components/TodoItem.js"
+        },
+        {
+          url: "http://example.com/index.js"
+        }
+      ];
+
+      const domain = "http://example.com/";
+      const sources = createSourcesList(testData);
+      const root = "/example.com/components";
+      // const root = "";
+      const tree = createNode("root", "", []);
+      sources.forEach(source => addToTree(tree, source, domain, root));
+      /*
+      - components
+      - TodoTextInput.js
+      - Header.js
+     */
```

Comment:
is it a working version, or is there something special to note here?

Reply:
Comments were there to remind the tree structure, but they are not required, i'll remove them